From f1b5f73a27817373d056af925fbe123d218697bb Mon Sep 17 00:00:00 2001
From: Peter Wu <lekensteyn@gmail.com>
Date: Tue, 1 Oct 2013 23:52:10 +0200
Subject: [PATCH 4/6] Add more PSK and Camellia ciphers

Notably, TLS_PSK_WITH_RC4_128_SHA was missing. This cipher suite is
supported by OpenSSL. The others are taken from the IANA page,
processed through generate-wireshark-cs[1].

 [1]: https://git.lekensteyn.nl/peter/wireshark-notes/tree/generate-wireshark-cs
---
 epan/dissectors/packet-ssl-utils.c | 87 ++++++++++++++++++++++++++++++++++++--
 1 file changed, 84 insertions(+), 3 deletions(-)

diff --git a/epan/dissectors/packet-ssl-utils.c b/epan/dissectors/packet-ssl-utils.c
index e478057..863e431 100644
--- a/epan/dissectors/packet-ssl-utils.c
+++ b/epan/dissectors/packet-ssl-utils.c
@@ -1790,6 +1790,9 @@ static SslCipherSuite cipher_suites[]={
     {25,KEX_DH,ENC_DES,8,64,40,DIG_SHA, SSL_CIPHER_MODE_CBC},   /* TLS_DH_anon_EXPORT_WITH_DES40_CBC_SHA */
     {26,KEX_DH,ENC_DES,8,64,64,DIG_SHA, SSL_CIPHER_MODE_CBC},   /* TLS_DH_anon_WITH_DES_CBC_SHA */
     {27,KEX_DH,ENC_3DES,8,192,192,DIG_SHA, SSL_CIPHER_MODE_CBC},   /* TLS_DH_anon_WITH_3DES_EDE_CBC_SHA */
+    {44,KEX_PSK,ENC_NULL,1,0,0,DIG_SHA, SSL_CIPHER_MODE_STREAM},   /* TLS_PSK_WITH_NULL_SHA */
+    {45,KEX_DH,ENC_NULL,1,0,0,DIG_SHA, SSL_CIPHER_MODE_STREAM},   /* TLS_DHE_PSK_WITH_NULL_SHA */
+    {46,KEX_RSA_PSK,ENC_NULL,1,0,0,DIG_SHA, SSL_CIPHER_MODE_STREAM},   /* TLS_RSA_PSK_WITH_NULL_SHA */
     {47,KEX_RSA,ENC_AES,16,128,128,DIG_SHA, SSL_CIPHER_MODE_CBC},   /* TLS_RSA_WITH_AES_128_CBC_SHA */
     {48,KEX_DH,ENC_AES,16,128,128,DIG_SHA, SSL_CIPHER_MODE_CBC},   /* TLS_DH_DSS_WITH_AES_128_CBC_SHA */
     {49,KEX_DH,ENC_AES,16,128,128,DIG_SHA, SSL_CIPHER_MODE_CBC},   /* TLS_DH_RSA_WITH_AES_128_CBC_SHA */
@@ -1834,9 +1837,18 @@ static SslCipherSuite cipher_suites[]={
     {135,KEX_DH,ENC_CAMELLIA256,16,256,256,DIG_SHA, SSL_CIPHER_MODE_CBC},   /* TLS_DHE_DSS_WITH_CAMELLIA_256_CBC_SHA */
     {136,KEX_DH,ENC_CAMELLIA256,16,256,256,DIG_SHA, SSL_CIPHER_MODE_CBC},   /* TLS_DHE_RSA_WITH_CAMELLIA_256_CBC_SHA */
     {137,KEX_DH,ENC_CAMELLIA256,16,256,256,DIG_SHA, SSL_CIPHER_MODE_CBC},   /* TLS_DH_anon_WITH_CAMELLIA_256_CBC_SHA */
-    {139,KEX_PSK,ENC_3DES,8,192,192,DIG_SHA, SSL_CIPHER_MODE_CBC},
-    {140,KEX_PSK,ENC_AES,16,128,128,DIG_SHA, SSL_CIPHER_MODE_CBC},
-    {141,KEX_PSK,ENC_AES256,16,256,256,DIG_SHA, SSL_CIPHER_MODE_CBC},
+    {138,KEX_PSK,ENC_RC4,1,128,128,DIG_SHA, SSL_CIPHER_MODE_STREAM},   /* TLS_PSK_WITH_RC4_128_SHA */
+    {139,KEX_PSK,ENC_3DES,8,192,192,DIG_SHA, SSL_CIPHER_MODE_CBC},   /* TLS_PSK_WITH_3DES_EDE_CBC_SHA */
+    {140,KEX_PSK,ENC_AES,16,128,128,DIG_SHA, SSL_CIPHER_MODE_CBC},   /* TLS_PSK_WITH_AES_128_CBC_SHA */
+    {141,KEX_PSK,ENC_AES256,16,256,256,DIG_SHA, SSL_CIPHER_MODE_CBC},   /* TLS_PSK_WITH_AES_256_CBC_SHA */
+    {142,KEX_DH,ENC_RC4,1,128,128,DIG_SHA, SSL_CIPHER_MODE_STREAM},   /* TLS_DHE_PSK_WITH_RC4_128_SHA */
+    {143,KEX_DH,ENC_3DES,8,192,192,DIG_SHA, SSL_CIPHER_MODE_CBC},   /* TLS_DHE_PSK_WITH_3DES_EDE_CBC_SHA */
+    {144,KEX_DH,ENC_AES,16,128,128,DIG_SHA, SSL_CIPHER_MODE_CBC},   /* TLS_DHE_PSK_WITH_AES_128_CBC_SHA */
+    {145,KEX_DH,ENC_AES256,16,256,256,DIG_SHA, SSL_CIPHER_MODE_CBC},   /* TLS_DHE_PSK_WITH_AES_256_CBC_SHA */
+    {146,KEX_RSA_PSK,ENC_RC4,1,128,128,DIG_SHA, SSL_CIPHER_MODE_STREAM},   /* TLS_RSA_PSK_WITH_RC4_128_SHA */
+    {147,KEX_RSA_PSK,ENC_3DES,8,192,192,DIG_SHA, SSL_CIPHER_MODE_CBC},   /* TLS_RSA_PSK_WITH_3DES_EDE_CBC_SHA */
+    {148,KEX_RSA_PSK,ENC_AES,16,128,128,DIG_SHA, SSL_CIPHER_MODE_CBC},   /* TLS_RSA_PSK_WITH_AES_128_CBC_SHA */
+    {149,KEX_RSA_PSK,ENC_AES256,16,256,256,DIG_SHA, SSL_CIPHER_MODE_CBC},   /* TLS_RSA_PSK_WITH_AES_256_CBC_SHA */
     {150,KEX_RSA,ENC_SEED,16,128,128,DIG_SHA, SSL_CIPHER_MODE_CBC},   /* TLS_RSA_WITH_SEED_CBC_SHA */
     {151,KEX_DH,ENC_SEED,16,128,128,DIG_SHA, SSL_CIPHER_MODE_CBC},   /* TLS_DH_DSS_WITH_SEED_CBC_SHA */
     {152,KEX_DH,ENC_SEED,16,128,128,DIG_SHA, SSL_CIPHER_MODE_CBC},   /* TLS_DH_RSA_WITH_SEED_CBC_SHA */
@@ -1855,6 +1867,24 @@ static SslCipherSuite cipher_suites[]={
     {165,KEX_DH,ENC_AES256,4,256,256,DIG_SHA384, SSL_CIPHER_MODE_GCM},   /* TLS_DH_DSS_WITH_AES_256_GCM_SHA384 */
     {166,KEX_DH,ENC_AES,4,128,128,DIG_SHA256, SSL_CIPHER_MODE_GCM},   /* TLS_DH_anon_WITH_AES_128_GCM_SHA256 */
     {167,KEX_DH,ENC_AES256,4,256,256,DIG_SHA384, SSL_CIPHER_MODE_GCM},   /* TLS_DH_anon_WITH_AES_256_GCM_SHA384 */
+    {168,KEX_PSK,ENC_AES,4,128,128,DIG_SHA256, SSL_CIPHER_MODE_GCM},   /* TLS_PSK_WITH_AES_128_GCM_SHA256 */
+    {169,KEX_PSK,ENC_AES256,4,256,256,DIG_SHA384, SSL_CIPHER_MODE_GCM},   /* TLS_PSK_WITH_AES_256_GCM_SHA384 */
+    {170,KEX_DH,ENC_AES,4,128,128,DIG_SHA256, SSL_CIPHER_MODE_GCM},   /* TLS_DHE_PSK_WITH_AES_128_GCM_SHA256 */
+    {171,KEX_DH,ENC_AES256,4,256,256,DIG_SHA384, SSL_CIPHER_MODE_GCM},   /* TLS_DHE_PSK_WITH_AES_256_GCM_SHA384 */
+    {172,KEX_RSA_PSK,ENC_AES,4,128,128,DIG_SHA256, SSL_CIPHER_MODE_GCM},   /* TLS_RSA_PSK_WITH_AES_128_GCM_SHA256 */
+    {173,KEX_RSA_PSK,ENC_AES256,4,256,256,DIG_SHA384, SSL_CIPHER_MODE_GCM},   /* TLS_RSA_PSK_WITH_AES_256_GCM_SHA384 */
+    {174,KEX_PSK,ENC_AES,16,128,128,DIG_SHA256, SSL_CIPHER_MODE_CBC},   /* TLS_PSK_WITH_AES_128_CBC_SHA256 */
+    {175,KEX_PSK,ENC_AES256,16,256,256,DIG_SHA384, SSL_CIPHER_MODE_CBC},   /* TLS_PSK_WITH_AES_256_CBC_SHA384 */
+    {176,KEX_PSK,ENC_NULL,1,0,0,DIG_SHA256, SSL_CIPHER_MODE_STREAM},   /* TLS_PSK_WITH_NULL_SHA256 */
+    {177,KEX_PSK,ENC_NULL,1,0,0,DIG_SHA384, SSL_CIPHER_MODE_STREAM},   /* TLS_PSK_WITH_NULL_SHA384 */
+    {178,KEX_DH,ENC_AES,16,128,128,DIG_SHA256, SSL_CIPHER_MODE_CBC},   /* TLS_DHE_PSK_WITH_AES_128_CBC_SHA256 */
+    {179,KEX_DH,ENC_AES256,16,256,256,DIG_SHA384, SSL_CIPHER_MODE_CBC},   /* TLS_DHE_PSK_WITH_AES_256_CBC_SHA384 */
+    {180,KEX_DH,ENC_NULL,1,0,0,DIG_SHA256, SSL_CIPHER_MODE_STREAM},   /* TLS_DHE_PSK_WITH_NULL_SHA256 */
+    {181,KEX_DH,ENC_NULL,1,0,0,DIG_SHA384, SSL_CIPHER_MODE_STREAM},   /* TLS_DHE_PSK_WITH_NULL_SHA384 */
+    {182,KEX_RSA_PSK,ENC_AES,16,128,128,DIG_SHA256, SSL_CIPHER_MODE_CBC},   /* TLS_RSA_PSK_WITH_AES_128_CBC_SHA256 */
+    {183,KEX_RSA_PSK,ENC_AES256,16,256,256,DIG_SHA384, SSL_CIPHER_MODE_CBC},   /* TLS_RSA_PSK_WITH_AES_256_CBC_SHA384 */
+    {184,KEX_RSA_PSK,ENC_NULL,1,0,0,DIG_SHA256, SSL_CIPHER_MODE_STREAM},   /* TLS_RSA_PSK_WITH_NULL_SHA256 */
+    {185,KEX_RSA_PSK,ENC_NULL,1,0,0,DIG_SHA384, SSL_CIPHER_MODE_STREAM},   /* TLS_RSA_PSK_WITH_NULL_SHA384 */
     {186,KEX_RSA,ENC_CAMELLIA128,16,128,128,DIG_SHA256, SSL_CIPHER_MODE_CBC},   /* TLS_RSA_WITH_CAMELLIA_128_CBC_SHA256 */
     {187,KEX_DH,ENC_CAMELLIA128,16,128,128,DIG_SHA256, SSL_CIPHER_MODE_CBC},   /* TLS_DH_DSS_WITH_CAMELLIA_128_CBC_SHA256 */
     {188,KEX_DH,ENC_CAMELLIA128,16,128,128,DIG_SHA256, SSL_CIPHER_MODE_CBC},   /* TLS_DH_RSA_WITH_CAMELLIA_128_CBC_SHA256 */
@@ -1908,6 +1938,57 @@ static SslCipherSuite cipher_suites[]={
     {49200,KEX_ECDH,ENC_AES256,4,256,256,DIG_SHA384, SSL_CIPHER_MODE_GCM},   /* TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384 */
     {49201,KEX_ECDH,ENC_AES,4,128,128,DIG_SHA256, SSL_CIPHER_MODE_GCM},   /* TLS_ECDH_RSA_WITH_AES_128_GCM_SHA256 */
     {49202,KEX_ECDH,ENC_AES256,4,256,256,DIG_SHA384, SSL_CIPHER_MODE_GCM},   /* TLS_ECDH_RSA_WITH_AES_256_GCM_SHA384 */
+    {49203,KEX_ECDH,ENC_RC4,1,128,128,DIG_SHA, SSL_CIPHER_MODE_STREAM},   /* TLS_ECDHE_PSK_WITH_RC4_128_SHA */
+    {49204,KEX_ECDH,ENC_3DES,8,192,192,DIG_SHA, SSL_CIPHER_MODE_CBC},   /* TLS_ECDHE_PSK_WITH_3DES_EDE_CBC_SHA */
+    {49205,KEX_ECDH,ENC_AES,16,128,128,DIG_SHA, SSL_CIPHER_MODE_CBC},   /* TLS_ECDHE_PSK_WITH_AES_128_CBC_SHA */
+    {49206,KEX_ECDH,ENC_AES256,16,256,256,DIG_SHA, SSL_CIPHER_MODE_CBC},   /* TLS_ECDHE_PSK_WITH_AES_256_CBC_SHA */
+    {49207,KEX_ECDH,ENC_AES,16,128,128,DIG_SHA256, SSL_CIPHER_MODE_CBC},   /* TLS_ECDHE_PSK_WITH_AES_128_CBC_SHA256 */
+    {49208,KEX_ECDH,ENC_AES256,16,256,256,DIG_SHA384, SSL_CIPHER_MODE_CBC},   /* TLS_ECDHE_PSK_WITH_AES_256_CBC_SHA384 */
+    {49209,KEX_ECDH,ENC_NULL,1,0,0,DIG_SHA, SSL_CIPHER_MODE_STREAM},   /* TLS_ECDHE_PSK_WITH_NULL_SHA */
+    {49210,KEX_ECDH,ENC_NULL,1,0,0,DIG_SHA256, SSL_CIPHER_MODE_STREAM},   /* TLS_ECDHE_PSK_WITH_NULL_SHA256 */
+    {49211,KEX_ECDH,ENC_NULL,1,0,0,DIG_SHA384, SSL_CIPHER_MODE_STREAM},   /* TLS_ECDHE_PSK_WITH_NULL_SHA384 */
+    {49266,KEX_ECDH,ENC_CAMELLIA128,16,128,128,DIG_SHA256, SSL_CIPHER_MODE_CBC},   /* TLS_ECDHE_ECDSA_WITH_CAMELLIA_128_CBC_SHA256 */
+    {49267,KEX_ECDH,ENC_CAMELLIA256,16,256,256,DIG_SHA384, SSL_CIPHER_MODE_CBC},   /* TLS_ECDHE_ECDSA_WITH_CAMELLIA_256_CBC_SHA384 */
+    {49268,KEX_ECDH,ENC_CAMELLIA128,16,128,128,DIG_SHA256, SSL_CIPHER_MODE_CBC},   /* TLS_ECDH_ECDSA_WITH_CAMELLIA_128_CBC_SHA256 */
+    {49269,KEX_ECDH,ENC_CAMELLIA256,16,256,256,DIG_SHA384, SSL_CIPHER_MODE_CBC},   /* TLS_ECDH_ECDSA_WITH_CAMELLIA_256_CBC_SHA384 */
+    {49270,KEX_ECDH,ENC_CAMELLIA128,16,128,128,DIG_SHA256, SSL_CIPHER_MODE_CBC},   /* TLS_ECDHE_RSA_WITH_CAMELLIA_128_CBC_SHA256 */
+    {49271,KEX_ECDH,ENC_CAMELLIA256,16,256,256,DIG_SHA384, SSL_CIPHER_MODE_CBC},   /* TLS_ECDHE_RSA_WITH_CAMELLIA_256_CBC_SHA384 */
+    {49272,KEX_ECDH,ENC_CAMELLIA128,16,128,128,DIG_SHA256, SSL_CIPHER_MODE_CBC},   /* TLS_ECDH_RSA_WITH_CAMELLIA_128_CBC_SHA256 */
+    {49273,KEX_ECDH,ENC_CAMELLIA256,16,256,256,DIG_SHA384, SSL_CIPHER_MODE_CBC},   /* TLS_ECDH_RSA_WITH_CAMELLIA_256_CBC_SHA384 */
+    {49274,KEX_RSA,ENC_CAMELLIA128,4,128,128,DIG_SHA256, SSL_CIPHER_MODE_GCM},   /* TLS_RSA_WITH_CAMELLIA_128_GCM_SHA256 */
+    {49275,KEX_RSA,ENC_CAMELLIA256,4,256,256,DIG_SHA384, SSL_CIPHER_MODE_GCM},   /* TLS_RSA_WITH_CAMELLIA_256_GCM_SHA384 */
+    {49276,KEX_DH,ENC_CAMELLIA128,4,128,128,DIG_SHA256, SSL_CIPHER_MODE_GCM},   /* TLS_DHE_RSA_WITH_CAMELLIA_128_GCM_SHA256 */
+    {49277,KEX_DH,ENC_CAMELLIA256,4,256,256,DIG_SHA384, SSL_CIPHER_MODE_GCM},   /* TLS_DHE_RSA_WITH_CAMELLIA_256_GCM_SHA384 */
+    {49278,KEX_DH,ENC_CAMELLIA128,4,128,128,DIG_SHA256, SSL_CIPHER_MODE_GCM},   /* TLS_DH_RSA_WITH_CAMELLIA_128_GCM_SHA256 */
+    {49279,KEX_DH,ENC_CAMELLIA256,4,256,256,DIG_SHA384, SSL_CIPHER_MODE_GCM},   /* TLS_DH_RSA_WITH_CAMELLIA_256_GCM_SHA384 */
+    {49280,KEX_DH,ENC_CAMELLIA128,4,128,128,DIG_SHA256, SSL_CIPHER_MODE_GCM},   /* TLS_DHE_DSS_WITH_CAMELLIA_128_GCM_SHA256 */
+    {49281,KEX_DH,ENC_CAMELLIA256,4,256,256,DIG_SHA384, SSL_CIPHER_MODE_GCM},   /* TLS_DHE_DSS_WITH_CAMELLIA_256_GCM_SHA384 */
+    {49282,KEX_DH,ENC_CAMELLIA128,4,128,128,DIG_SHA256, SSL_CIPHER_MODE_GCM},   /* TLS_DH_DSS_WITH_CAMELLIA_128_GCM_SHA256 */
+    {49283,KEX_DH,ENC_CAMELLIA256,4,256,256,DIG_SHA384, SSL_CIPHER_MODE_GCM},   /* TLS_DH_DSS_WITH_CAMELLIA_256_GCM_SHA384 */
+    {49284,KEX_DH,ENC_CAMELLIA128,4,128,128,DIG_SHA256, SSL_CIPHER_MODE_GCM},   /* TLS_DH_anon_WITH_CAMELLIA_128_GCM_SHA256 */
+    {49285,KEX_DH,ENC_CAMELLIA256,4,256,256,DIG_SHA384, SSL_CIPHER_MODE_GCM},   /* TLS_DH_anon_WITH_CAMELLIA_256_GCM_SHA384 */
+    {49286,KEX_ECDH,ENC_CAMELLIA128,4,128,128,DIG_SHA256, SSL_CIPHER_MODE_GCM},   /* TLS_ECDHE_ECDSA_WITH_CAMELLIA_128_GCM_SHA256 */
+    {49287,KEX_ECDH,ENC_CAMELLIA256,4,256,256,DIG_SHA384, SSL_CIPHER_MODE_GCM},   /* TLS_ECDHE_ECDSA_WITH_CAMELLIA_256_GCM_SHA384 */
+    {49288,KEX_ECDH,ENC_CAMELLIA128,4,128,128,DIG_SHA256, SSL_CIPHER_MODE_GCM},   /* TLS_ECDH_ECDSA_WITH_CAMELLIA_128_GCM_SHA256 */
+    {49289,KEX_ECDH,ENC_CAMELLIA256,4,256,256,DIG_SHA384, SSL_CIPHER_MODE_GCM},   /* TLS_ECDH_ECDSA_WITH_CAMELLIA_256_GCM_SHA384 */
+    {49290,KEX_ECDH,ENC_CAMELLIA128,4,128,128,DIG_SHA256, SSL_CIPHER_MODE_GCM},   /* TLS_ECDHE_RSA_WITH_CAMELLIA_128_GCM_SHA256 */
+    {49291,KEX_ECDH,ENC_CAMELLIA256,4,256,256,DIG_SHA384, SSL_CIPHER_MODE_GCM},   /* TLS_ECDHE_RSA_WITH_CAMELLIA_256_GCM_SHA384 */
+    {49292,KEX_ECDH,ENC_CAMELLIA128,4,128,128,DIG_SHA256, SSL_CIPHER_MODE_GCM},   /* TLS_ECDH_RSA_WITH_CAMELLIA_128_GCM_SHA256 */
+    {49293,KEX_ECDH,ENC_CAMELLIA256,4,256,256,DIG_SHA384, SSL_CIPHER_MODE_GCM},   /* TLS_ECDH_RSA_WITH_CAMELLIA_256_GCM_SHA384 */
+    {49294,KEX_PSK,ENC_CAMELLIA128,4,128,128,DIG_SHA256, SSL_CIPHER_MODE_GCM},   /* TLS_PSK_WITH_CAMELLIA_128_GCM_SHA256 */
+    {49295,KEX_PSK,ENC_CAMELLIA256,4,256,256,DIG_SHA384, SSL_CIPHER_MODE_GCM},   /* TLS_PSK_WITH_CAMELLIA_256_GCM_SHA384 */
+    {49296,KEX_DH,ENC_CAMELLIA128,4,128,128,DIG_SHA256, SSL_CIPHER_MODE_GCM},   /* TLS_DHE_PSK_WITH_CAMELLIA_128_GCM_SHA256 */
+    {49297,KEX_DH,ENC_CAMELLIA256,4,256,256,DIG_SHA384, SSL_CIPHER_MODE_GCM},   /* TLS_DHE_PSK_WITH_CAMELLIA_256_GCM_SHA384 */
+    {49298,KEX_RSA_PSK,ENC_CAMELLIA128,4,128,128,DIG_SHA256, SSL_CIPHER_MODE_GCM},   /* TLS_RSA_PSK_WITH_CAMELLIA_128_GCM_SHA256 */
+    {49299,KEX_RSA_PSK,ENC_CAMELLIA256,4,256,256,DIG_SHA384, SSL_CIPHER_MODE_GCM},   /* TLS_RSA_PSK_WITH_CAMELLIA_256_GCM_SHA384 */
+    {49300,KEX_PSK,ENC_CAMELLIA128,16,128,128,DIG_SHA256, SSL_CIPHER_MODE_CBC},   /* TLS_PSK_WITH_CAMELLIA_128_CBC_SHA256 */
+    {49301,KEX_PSK,ENC_CAMELLIA256,16,256,256,DIG_SHA384, SSL_CIPHER_MODE_CBC},   /* TLS_PSK_WITH_CAMELLIA_256_CBC_SHA384 */
+    {49302,KEX_DH,ENC_CAMELLIA128,16,128,128,DIG_SHA256, SSL_CIPHER_MODE_CBC},   /* TLS_DHE_PSK_WITH_CAMELLIA_128_CBC_SHA256 */
+    {49303,KEX_DH,ENC_CAMELLIA256,16,256,256,DIG_SHA384, SSL_CIPHER_MODE_CBC},   /* TLS_DHE_PSK_WITH_CAMELLIA_256_CBC_SHA384 */
+    {49304,KEX_RSA_PSK,ENC_CAMELLIA128,16,128,128,DIG_SHA256, SSL_CIPHER_MODE_CBC},   /* TLS_RSA_PSK_WITH_CAMELLIA_128_CBC_SHA256 */
+    {49305,KEX_RSA_PSK,ENC_CAMELLIA256,16,256,256,DIG_SHA384, SSL_CIPHER_MODE_CBC},   /* TLS_RSA_PSK_WITH_CAMELLIA_256_CBC_SHA384 */
+    {49306,KEX_ECDH,ENC_CAMELLIA128,16,128,128,DIG_SHA256, SSL_CIPHER_MODE_CBC},   /* TLS_ECDHE_PSK_WITH_CAMELLIA_128_CBC_SHA256 */
+    {49307,KEX_ECDH,ENC_CAMELLIA256,16,256,256,DIG_SHA384, SSL_CIPHER_MODE_CBC},   /* TLS_ECDHE_PSK_WITH_CAMELLIA_256_CBC_SHA384 */
     {-1, 0,0,0,0,0,0, 0}
 };
 
-- 
1.8.4

